### DISCUSSION TITLE
Right to Left Languages Support

### DISCUSSION THREAD

**Discussion by hishamco** [oqtane/oqtane.framework#1266] [Ideas]
While Oqtane Framework already has a success story with localization, there are still some of missing localization features one of the is to support RTL languages, this could be done by:

- Allow the Oqtane host respect the direction of the current culture
- Allow themes to support both LTR & RTL, this include
  - Change styles
  - Change controls
  - Chane layouts

I'm glad to implement this coz my language is RTL ;)

**Comment by ajahangard** [comment]
I tried this on _Host.cshtml and everything is working fine.

`@using System.Globalization;`
`<html dir="@(CultureInfo.CurrentUICulture.TextInfo.IsRightToLeft?"rtl":"ltr")" lang="@CultureInfo.CurrentUICulture.TwoLetterISOLanguageName">`

I also  conditionally added Bootstrap rtl to theme resources.

**Reply by hishamco** [reply]
I did something similar yesterday, so I decided to share my thoughts with the community before the implementation

**Comment by ajahangard** [comment]
I Have a recommendation about language management.
I think it is a good feature to be able to set language at page level.
Currently the language is set globally but if we want to have a multi language site using oqtane it is good to be able to set language at page level.
any ideas?

**Reply by hishamco** [reply]
Could you please start a new discussion about "Language Management", so this will be focus on RTL

Thanks
